.github/workflows/ci-arm64.yml

```diff
@@ -115,5 +157,95 @@ jobs:
         ./Configure
         make
         make test
-        .github/workflows/scripts/benchmark.sh > benchmark.txt 2>&1
-        cat benchmark.txt
+        .github/workflows/scripts/benchmark.sh > benchmark_aarch64_gcc.txt 2>&1
+        cat benchmark_aarch64_gcc.txt
+
+    - name: Archive openssl_linux_aarch64_gcc_benchmark.zip
+      uses: actions/upload-artifact@v3
+      if: ${{ inputs.benchmark }} || ${{ inputs.benchmark_snapshot }}
+      with:
+        name: openssl_linux_aarch64_gcc_benchmark.zip
+        path: benchmark_aarch64_gcc.txt
+        retention-days: 3
+
+  verify-benchmark-regression:
+    if: ${{ inputs.verify_benchmark }}
+    needs: [build-openssl-windows-arm64-clangcl, build-openssl-linux-aarch64-gcc]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Git checkout
+      uses: actions/checkout@v3
+
+    - name: Dowload openssl_linux_aarch64_gcc_benchmark.zip
+      uses: actions/download-artifact@v3
+      with:
+        name: openssl_linux_aarch64_gcc_benchmark.zip
+
+    - name: Dowload openssl_windows_arm64_clangcl_benchmark.zip
+      uses: actions/download-artifact@v3
+      with:
+        name: openssl_windows_arm64_clangcl_benchmark.zip
+
+    - name: Verify benchmark regression
+      run: |
+        set -x
+        git fetch origin ${{ github.event.repository.default_branch }}
+        git checkout origin/${{ github.event.repository.default_branch }} -- .github/workflows/scripts
+        . .github/workflows/scripts/benchmark.sh verify_benchmark_regression
+
+  create-benchmark-snapshot:
+    if: ${{ inputs.benchmark_snapshot }}
+    needs: [build-openssl-windows-arm64-clangcl, build-openssl-linux-aarch64-gcc]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Git checkout
+      uses: actions/checkout@v3
+
+    - name: Dowload openssl_linux_aarch64_gcc_benchmark.zip
+      uses: actions/download-artifact@v3
+      with:
+        name: openssl_linux_aarch64_gcc_benchmark.zip
+
+    - name: Dowload openssl_windows_arm64_clangcl_benchmark.zip
+      uses: actions/download-artifact@v3
+      with:
+        name: openssl_windows_arm64_clangcl_benchmark.zip
+
+    - name: Create PR for a benchmark snapshot
+      run: |
+        set -x
+        git config --global user.name "$GITHUB_ACTOR"
```

Comment:
Since this will need to be executed repeatedly (as one never know in what state the runner is), it not needed to store the user name and e-mail to the global git config. Saving it to local repository config would be bettter/safer.

Reply:
Changed